src/js_app.hpp

```diff
@@ -30,14 +30,34 @@
 #include "js_network_transport.hpp"
 #include "js_email_password_auth.hpp"
 
+#include <forward_list>
+
 using SharedApp = std::shared_ptr<realm::app::App>;
 using SharedUser = std::shared_ptr<realm::SyncUser>;
+using AppToken = realm::Subscribable<realm::app::App>::Token;
 
 namespace realm {
 namespace js {
 
 template <typename T>
-class AppClass : public ClassDefinition<T, SharedApp> {
+class App {
+public:
+    App(const SharedApp& l)
+        : m_app(l)
+    {
+    }
+
+    App(const App&) = delete;
+    App& operator=(const App&) = delete;
+
+    using CallbackTokenPair = std::pair<Protected<typename T::Function>, AppToken>;
+    std::forward_list<CallbackTokenPair> m_notification_tokens;
+
+    SharedApp m_app;
```

Comment:
I was having issues that went away by implementing it with a member.  That being said, it's possible that it could work extending the `shared_ptr` directly.  Maybe a second look would be worthwhile, but this time using `->get()` to retrieve a reference to `app` instead of dereferencing directly.

Reply:
That didn't work the way I had hoped.  Found some reading on inheriting `shared_ptr` though..
https://stackoverflow.com/a/30309840/192046